manifests/agent/windows.pp

```diff
@@ -12,62 +17,74 @@
     $url = "https://s3.amazonaws.com/download.distelli.com/distelli.Windows-x86/${archive}"
   }
 
-  $workdir = 'C:/Program Files/Distelli'
+  if $pipelines::agent::user_groups {
+    $user_groups = ['Users','Administrators'] + $pipelines::agent::user_groups
+  }
+  else {
+    $user_groups = ['Users','Administrators']
+  }
+
+  user { 'distelli':
+    ensure     => present,
+    comment    => 'Puppet Pipelines User',
+    home       => $pipelines::agent::user_home,
+    groups     => $user_groups,
+    password   => $pipelines::agent::user_password,
+    managehome => true,
+  }
 
-  archive { "${tempdir}/${archive}" :
+  archive { "${tempdir}/${archive}":
     source       => $url,
-    # cleanup      => false,
     creates      => "${tempdir}/distelli",
     extract      => true,
     extract_path => $tempdir,
     require      => User['distelli'],
   }
 
-  file { $workdir :
-    ensure  => directory,
-    owner   => 'distelli',
-    group   => 'Administrators',
-    require => User['distelli'],
+  file { $workdir:
+    ensure => directory,
+    owner  => 'distelli',
+    group  => 'Administrators',
   }
 
-  # Requires fqdn_rand_string function from puppetlabs/stdlib
-  $distelli_exec = "distelli-${fqdn_rand_string(5)}.exe"
-
-  exec { 'Copy executable' :
+  exec { 'pipelines::agent::windows Copy executable':
     command   => "Copy-Item ${tempdir}/distelli \$ENV:ProgramFiles/Distelli/${distelli_exec}",
-    unless    => "If (Test-Path -Path \$ENV:ProgramFiles/Distelli/${distelli_exec}) { exit 0 } else { exit 1}",
+    unless    => "If (Test-Path -Path \$ENV:ProgramFiles/Distelli/${distelli_exec}) { exit 0 } else { exit 1 }",
     provider  => powershell,
```

Comment:
Since the use of PowerShell here is so limited, I would recommend removing the PowerShell module dependency altogether and just stick to basic `exec` against `cmd`.  Can't this resource just be a `File` anyhow?

Reply:
That would solve the windows testing issue too I mentioned earlier (I think)